src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -119,6 +120,70 @@ internal static bool TryParseOption(string arg, out string name, out string valu
             return true;
         }
 
+        internal ErrorLogOptions ParseErrorLogOptions(
+            string arg,
+            IList<Diagnostic> diagnostics,
+            string baseDirectory,
+            out bool diagnosticAlreadyReported)
+        {
+            diagnosticAlreadyReported = false;
+
+            bool success = true;
+            string path = null;
+            SarifVersion sarifVersion = SarifVersion.Default;
+
+            string[] parts = ParseSeparatedStrings(arg, s_pathSeparators, StringSplitOptions.RemoveEmptyEntries).ToArray();
```

Comment:
Thanks, fixed.

---
In reply to: [322446314](https://github.com/dotnet/roslyn/pull/37148#discussion_r322446314) [](ancestors = 322446314)

Reply:
I really don't see the need for either.

From the top of my head (and without using spans):

```csharp
internal ErrorLogOptions ParseErrorLogOptions(
    string arg,
    IList<Diagnostic> diagnostics,
    string baseDirectory,
    out bool diagnosticAlreadyReported)
{
    diagnosticAlreadyReported = false;

    var partsEnumerator = ParseSeparatedStrings(arg, s_pathSeparators, StringSplitOptions.RemoveEmptyEntries).GetEnumerator();

    if (!partsEnumerator.MoveNext() || !string.IsNullOrEmpty(partsEnumerator.Current))
    {
        return null;
    }
    
    var path = partsEnumerator.Current;

    if (!partsEnumerator.MoveNext())
    {
        return null;
    }

    var version = partsEnumerator.Current;

    if (partsEnumerator.MoveNext())
    {
        return null;
    }

    if (!(ParseGenericPathToFile(path, diagnostics, baseDirectory) is string genericPath))
    {
        // ParseGenericPathToFile already reported the failure, so the caller should not
        // report its own failure.
        diagnosticAlreadyReported = true;

        return null;
    }

    const int equalsIndex = 7; // "version".Length
    if (!(
            partsEnumerator.Current.Length > equalsIndex + 1 && partsEnumerator.Current[equalsIndex + 1] == '=' &&
            partsEnumerator.Current.Substring(0, equalsIndex).Equals("version", StringComparison.OrdinalIgnoreCase) &&
            SarifVersionFacts.TryParse(partsEnumerator.Current.Substring(equalsIndex + 2), out var sarifVersion)
        ))
    {
        return null;
    }

    return new ErrorLogOptions(path, sarifVersion);
}
```

Reply:
I like your code. My concern is that the next time we introduce another qualifier on the `errorlog` option -- for example, the `allrules` qualifier you asked for, we won't be able to just step through the enumeration, because the qualifiers (`version`, `errorlog`, etc.) should be able to occur in any order.

However, I will adopt your `return null;` rather than `success == false;`. I wasn't sure of the Roslyn convention for "early return". I now see that it is used elsewhere in the Roslyn code.

---
In reply to: [323110284](https://github.com/dotnet/roslyn/pull/37148#discussion_r323110284) [](ancestors = 323110284)

Reply:
Another qualifier, another `MoveNext()`. 😄

Reply:
Ok, I `yield` :)

---
In reply to: [323477150](https://github.com/dotnet/roslyn/pull/37148#discussion_r323477150) [](ancestors = 323477150)